### PR TITLE
add: basic docs on alpha feature imageallowrules

### PR DESCRIPTION
**Pull request by iwilltry42**
Signed-off-by: Thorsten Klein <tk@thklein.io>### Checklist
- [ ] The title of this PR would make a good line in Acorn's Release Note's Changelog
- [ ] The title of this PR ends with a link to the main issue being address in parentheses, like: `This is a title (#1216)`. [Here's an example](https://github.com/acorn-io/acorn/pull/1199)
- [ ] All relevant issues are referenced in the PR description. *NOTE: don't use [GitHub keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) that auto-close issues*
- [ ] Commits follow [contributing guidance](https://github.com/acorn-io/acorn/blob/main/CONTRIBUTING.md#commits)
- [ ] Automated tests added to cover the changes. If tests couldn't be added, an explanation is provided in the Verification and Testing section
- [ ] Changes to user-facing functionality, API, CLI, and upgrade impacts are clearly called out in PR description
- [ ] PR has at least two approvals before merging (or a reasonable exception, like it's just a docs change)

### CODE REVIEW

File: docs/docs/50-running/80-alpha-image-allow-rules.md [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+To do that, you have to create an `ImageAllowRule` resource.
+If the app image is allowed by a single IAR in your project, it's good to run.
+
+**Note**: Removing an IAR won't stop your running app, but will update the `image-allowed` status condition on the app.
+
+## What makes up an ImageAllowRule
+
+Currently, IARs have two parts:
+
+1. The `images` scope (must-have), denoting, what images the rule applies to. It uses the same syntax as the auto-upgrade pattern. Examples below.
```
Comment (g-linville) [inline docs/docs/50-running/80-alpha-image-allow-rules.md:20]:
```suggestion
1. The `images` scope (must-have), denoting what images the rule applies to. It uses the same syntax as the auto-upgrade pattern. Examples below.
```

File: docs/docs/50-running/80-alpha-image-allow-rules.md [hunk 2]
```diff
@@ -0,0 +1,154 @@
+# [Alpha Feature] ImageAllowRules (IARs)
+
+ImageAllowRules are an alpha-feature of Acorn, currently hidden behind a feature flag.
+To enable it in your Acorn installation, use `acorn isntall --features image-allow-rules=true` (**Beware**: first read the rest of this page before enabling it, as it's quite disruptive).
+
+## How ImageAllowRules work
+
+The principle behind IARs is to make your cluster more secure.
+In fact, secure by default, once it's enabled. No additional policies needed.
```
Comment (g-linville) [inline docs/docs/50-running/80-alpha-image-allow-rules.md:9]:
```suggestion
The principle behind IARs is to make your cluster more secure by default.
```

File: docs/docs/50-running/80-alpha-image-allow-rules.md [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+2. The `signatures` rules (optional) define a set of image signatures and annotations on those signatures to make sure that an image was actually approved by someone or something, e.g. by your QA team. We're using [sigstore/cosign](https://docs.sigstore.dev/cosign/installation/) for everything related to signatures.
+
+## Example
+
+```yaml
+apiVersion: api.acorn.io/v1
+kind: ImageAllowRule
+metadata:
+  name: example-iar
+  namespace: acorn # your project namespace
```
Comment (g-linville) [inline docs/docs/50-running/80-alpha-image-allow-rules.md:30]: I think this is slightly clearer for any Acorn user's who don't know that project name == project namespace.

```suggestion
  namespace: acorn # your project name
```

File: docs/docs/50-running/80-alpha-image-allow-rules.md [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+            operator: In # In, NotIn, Exists, DoesNotExist
+            values:
+              - passed
+              - ok
+```
+
+## About Signatures
+
+To sign an image, you can use [sigstore/cosign](https://docs.sigstore.dev/cosign/installation/) via the CLI.
+You download or build an Acorn image, you sign it with cosign and optionally annotate the signature, then you upload it to some OCI registry.
```
Comment (g-linville) [inline docs/docs/50-running/80-alpha-image-allow-rules.md:57]:
```suggestion
You can download or build an Acorn image, sign it with cosign and optionally annotate the signature, then upload it to some OCI registry.
```

File: docs/docs/50-running/80-alpha-image-allow-rules.md [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+Now when you try to run that in a protected cluster and the image is in scope of an IAR, Acorn will check the signature, make sure that it matches the provided public keys and that matching signatures also have the required annotations.
```
Comment (g-linville) [inline docs/docs/50-running/80-alpha-image-allow-rules.md:58]:
```suggestion
When you try to run the image in a protected cluster and the image is in scope of an IAR, Acorn will check the signature, make sure that it matches the provided public keys and that matching signatures also have the required annotations.
```

File: docs/docs/50-running/80-alpha-image-allow-rules.md [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+Only if all of this is true, we let the image pass.
```
Comment (g-linville) [inline docs/docs/50-running/80-alpha-image-allow-rules.md:59]:
```suggestion
Acorn will only run the image if everything matches.
```

**Review by g-linville — COMMENTED** [review]
Nice tutorial, just a few nits for conciseness and clarity

**Review by njhale — APPROVED** [review]

**Review by thedadams — APPROVED** [review]

**Review by cloudnautique — APPROVED** [review]

**Pull request merged**